shared/src/main/scala/org/adridadou/openlaw/values/ContractDefinition.scala

```diff
@@ -15,7 +15,11 @@ final case class ContractDefinition(
     templates: Map[TemplateSourceIdentifier, TemplateId] = Map.empty,
     parameters: TemplateParameters,
     paragraphs: Map[Int, ParagraphEdits] = Map.empty,
-    creationDate: LocalDateTime = LocalDateTime.now
+    creationDate: LocalDateTime = LocalDateTime.now,
+
+    // If the contract is instead formed by a PDF file rather than an openlaw template, we include the hash of the PDF
+    // file
+    pdfHash: Option[String] = None
```

Comment:
I usually prefer when we type things. We're going to use the same hashing we have for Ethereum so the type should be EthereumHash IMO